cmd/renderer/main.go

```
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"io/ioutil"
+	"os"
+	"strings"
+	"text/template"
+)
+
+func main() {
+
+	templateFile := ""
+	variablesFile := ""
+
+	flag.StringVar(&templateFile, "t", "", "template file")
+	flag.StringVar(&variablesFile, "v", "", "variables file")
+
+	flag.Parse()
+
+	// If no template file given, then error out
+	if len(templateFile) == 0 {
+		panic(errors.New("file is empty"))
+	}
+
+	// Read contents of template file into tmpl
+	tmpl, err := ioutil.ReadFile(templateFile)
+	if err != nil {
+		panic(err)
+	}
+
+	// Read contents of variables file into vars
+	vars, err := ioutil.ReadFile(variablesFile)
+	if err != nil {
+		panic(err)
+	}
+
+	ctx := map[string]string{}
+
+	// Adds vairables from file into context
+	for _, x := range strings.Split(string(vars), "\n") {
+		if len(x) > 0 {
+			pair := strings.SplitAfterN(x, "=", 2)
+			ctx[pair[0][0:len(pair[0])-1]] = pair[1]
+		}
+	}
+
+	// Adds environment vairables to context
+	for _, x := range os.Environ() {
+		pair := strings.SplitAfterN(x, "=", 2)
+		ctx[pair[0][0:len(pair[0])-1]] = pair[1]
+	}
+
+	// Adds command line arguments to context
+	for _, x := range flag.Args() {
+		pair := strings.SplitAfterN(x, "=", 2)
+		ctx[pair[0][0:len(pair[0])-1]] = pair[1]
+	}
+
+	template.Must(template.New("main").Parse(string(tmpl))).Execute(os.Stdout, ctx)
```

by default the template will put `no value` if there isn't a defined value. We should really error out if the value is missing. 

https://golang.org/pkg/text/template/#Template.Option

---

Good idea!